pkg/sql/schemachanger/scbuild/internal/scbuildstmt/drop_index.go

```diff
@@ -270,15 +270,18 @@ func maybeDropDependentFKConstraints(
 	// dropDependentFKConstraint is a helper function that drops a dependent
 	// FK constraint with ID `fkConstraintID`.
 	dropDependentFKConstraint := func(fkConstraintID catid.ConstraintID) {
-		b.BackReferences(tableID).Filter(hasConstraintIDAttrFilter(fkConstraintID)).
+		b.BackReferences(tableID).Filter(containsDescIDFilter(tableID)).Filter(hasConstraintIDAttrFilter(fkConstraintID)).
```

Comment:
I don't think this is necessary(?) since we already filtered FKs earlier (below) to only drop FKs inbound to this table.

Reply:
Since we are getting back-references again, isn't there a danger of accidentally operating on the wrong table with the same constraint ID? I think this might be too relaxed still actually

Reply:
that's a good point! Yes, we should filter by descID